bin/libs/zos-jes.ts

```diff
@@ -0,0 +1,253 @@
+        if (hasplines && hasplines.length > 0) {
+          const jobline = hasplines[0];
+          const nameIndex = jobline.indexOf('JOB(');
+          const ccIndex = jobline.indexOf('CC=(');
+          jobname = jobline.substring(nameIndex+4, jobline.indexOf(')', nameIndex));
+          const cc = jobline.substring(ccIndex+4, jobline.indexOf(')', ccIndex)).split(',');
+          jobcctext = cc[0];
+          if (cc.length > 1) {
+            const equalSplit = cc[1].split('=');
+            if (equalSplit.length > 1) {
+              jobcccode = equalSplit[1];
+            }
+          }
+          common.printTrace(`  * Job (${jobname}) status is ${jobcctext},RC=${jobcccode}`);
+          if ((jobcctext && jobcctext.length > 0) || (jobcccode && jobcccode.length > 0)) {
+            // job have CC state
+            break;
+          }
+        }
+      } catch (e) {
+        break;
+      }
+    }
+  }
+  common.printTrace(`  * Job status check done at ${new Date().toString()}.`);
+
+  if (jobcctext || jobcccode) {
+    common.printDebug(`  * Job (${jobname}) exits with code ${jobcccode} (${jobcctext}).`);
+    if (jobcccode == "0") {
+      return {jobcctext, jobcccode, jobname, rc: 0};
+    } else {
+      // ${jobcccode} could be greater than 255
+      return {jobcctext, jobcccode, jobname, rc: 2};
+    }
+  } else if (is_jes3) {
+    common.printTrace(`  * Cannot determine job complete code. Please check job log manually.`);
+    return {jobcctext, jobcccode, jobname, rc: 0};
+  } else {
+    common.printError(`  * Job (${jobname? jobname : jobid}) doesn't finish within max waiting period.`);
+    return {jobcctext, jobcccode, jobname, rc: 1};
+  }
+}
+
+export function printAndHandleJcl(jclLocationOrContent: string, jobName: string, jcllib: string, prefix: string, removeJclOnFinish?: boolean, continueOnFailure?: boolean, jclIsContent?: boolean){
+  const jclContents = jclIsContent ? jclLocationOrContent : shell.execOutSync('sh', '-c', `cat "${jclLocationOrContent}" 2>&1`).out;
+
+  let jobHasFailures = false;
+  if (jclIsContent) {
+    removeJclOnFinish = false;
+  }
+
+  common.printMessage(`Template JCL: ${prefix}.SZWESAMP(${jobName}) , Executable JCL: ${jcllib}(${jobName})`);
+  common.printMessage(`--- JCL Content ---`);
+  common.printMessage(jclContents);
+  common.printMessage(`--- End of JCL ---`);
+
+  common.printTrace('  * zos-jes.printAndHanleJcl');
+  common.printTrace('    * JCL Lines Length');
+  const jclContentsSplit = jclContents.split("\n");
+  for (let jclLine in jclContentsSplit) {
+      const tracePad = 6;
+      common.printTrace(`${jclContentsSplit[jclLine].length.toString().padStart(tracePad, ' ')}: ${jclContentsSplit[jclLine]}`);
+      if (jclContentsSplit[jclLine].length > 71) {
+          common.printTrace(`${' '.repeat(tracePad + 2)}${'^'.repeat(jclContentsSplit[jclLine].length)}`);
+      }
+  }
+  common.printTrace('    * JCL Lines Length');
```

Comment:
This trace can be deleted before merging. But I would like to keep it for the testing now.